sr/robot/utils.py

```diff
@@ -216,14 +216,31 @@ def ensure_atexit_on_term() -> None:
         # this is sufficient for `atexit` to trigger, so do nothing.
         return
 
-    def handle_signal(handled_signum: int, frame: FrameType | None) -> None:
+    def handle_signal(handled_signum: int, frame: Optional[FrameType]) -> None:
         """
         Handle the given signal by outputting some text and terminating the process.
-
         This will trigger `atexit`.
         """
         logger.info(signal.strsignal(handled_signum))
         exit(1)
 
     # Add the null-ish signal handler
     signal.signal(signal.SIGTERM, handle_signal)
+
+
+def list_ports() -> None:
+    """
+    Print a list of all connected USB serial ports.
+
+    Output is in the format:
+    ```
+    VID:PID - Manufacturer - Product - Serial Number
+    ```
+    """
+    for serial_port in comports():
+        if serial_port.vid and serial_port.pid:
+            # Only print USB serial ports
+            print(
```

Comment:
Should we add a line before this print to make it more clear what is being printed
e.g. 
`print("VID:PID - Manufacturer - Product - Serial Number")`